src/xtgeo/grid3d/_gridprop_lowlevel.py

```diff
@@ -54,15 +68,21 @@ def update_values_from_carray(self, carray, dtype, delete=False):
     values = np.asanyarray(values, order="C")
 
     # make it float64 or whatever(?) and mask it
-    self._values = values
+    self._values = values # type: ignore
     self.mask_undef()
```

Comment:
Replacing this with the setter `self.values = values` would give a partial resolution to this typing -- partial because I see that the setter should actually accept an `ArrayLike`, not just masked arrays (it will convert the argument into a masked array). Haven't looked but maybe could remove the need to call `mask_undef` too.

https://github.com/equinor/xtgeo/blob/08db8fbd04428b057f71a14d0ec799aca550e7a8/src/xtgeo/grid3d/grid_property.py#L750-L751

Reply:
I tried ArrayLike, but i got so many errors that i felt like it was something to address at a later stage.